openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright 2024 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { isUndefined, omitBy, round } from 'lodash';
+import { TestCaseChartDataType } from '../../components/Database/Profiler/ProfilerDashboard/profilerDashboard.interface';
+import { COLORS } from '../../constants/profiler.constant';
+import { Thread } from '../../generated/entity/feed/thread';
+import {
+  TestCaseParameterValue,
+  TestCaseResult,
+} from '../../generated/tests/testCase';
+
+export type PrepareChartDataType = {
+  testCaseParameterValue: TestCaseParameterValue[];
+  testCaseResults: TestCaseResult[];
+  entityThread: Thread[];
+};
+
+export const prepareChartData = ({
+  testCaseParameterValue,
+  testCaseResults,
+  entityThread,
+}: PrepareChartDataType) => {
+  const params =
+    testCaseParameterValue?.length >= 2 ? testCaseParameterValue : [];
+  const dataPoints: TestCaseChartDataType['data'] = [];
+  const yValues = params.reduce((acc, curr, i) => {
+    return { ...acc, [`y${i + 1}`]: parseInt(curr.value ?? '') };
```

Comment:
`parseInt(curr.value ?? '')` here uses a string as a fallback for parseInt, resulting in NaN if the value is null or undefined.

Reply:
Yes, it's expected to have such value so the chart doesn't draw anything!